aten/src/ATen/native/cuda/AdaptiveMaxPooling2d.cu

```diff
@@ -0,0 +1,472 @@
+
+  // the jit sometimes passes output_size.size() == 1
+  AT_CHECK(output_size.size() == 1 || output_size.size() == 2,
+    "adaptive_max_pool2d: internal error: output_size.size() must be 1 or 2");
+
+  int64_t osizeH = output_size[0];
+  int64_t osizeW = output_size.size() == 1 ? output_size[0] : output_size[1];
+
+  if (input.ndimension() == 3) {
+    int64_t sizeD  = input.size(0);
+    int64_t isizeH = input.size(1);
+    int64_t isizeW = input.size(2);
+
+    int64_t istrideD = input.stride(0);
+    int64_t istrideH = input.stride(1);
+    int64_t istrideW = input.stride(2);
+
+    AT_DISPATCH_FLOATING_TYPES_AND_HALF(input.scalar_type(),
+      "adaptive_max_pool2d_cuda",
+      [&] {
+        output.resize_({sizeD, osizeH, osizeW});
+        indices.resize_({sizeD, osizeH, osizeW});
+
+        scalar_t *input_data = input.data<scalar_t>();
+        scalar_t *output_data = output.data<scalar_t>();
+        int64_t *indices_data = indices.data<int64_t>();
+
+        // cuda blocks & threads:
+        int blocksH = (int)(16L / sizeD);
+        blocksH = blocksH < 1 ? 1 : blocksH;
+        dim3 blocks(sizeD, blocksH);
+        dim3 threads(32, 8);
+
+        // run maxpool kernel
+        adaptivemaxpool <<<blocks, threads, 0, at::cuda::getCurrentCUDAStream()>>> (
+                                   input_data, output_data,
+                                   indices_data,
+                                   isizeH, isizeW, osizeH, osizeW,
+                                   istrideD, istrideH, istrideW);
+      }
+    );
+    THCudaCheck(cudaGetLastError());
+
+  } else {
+    Tensor input_ = input.contiguous();
+    int64_t sizeB  = input_.size(0);
+    int64_t sizeD  = input_.size(1);
+    int64_t isizeH = input_.size(2);
+    int64_t isizeW = input_.size(3);
+
+    int64_t istrideD = input_.stride(1);
+    int64_t istrideH = input_.stride(2);
+    int64_t istrideW = input_.stride(3);
+
+    AT_DISPATCH_FLOATING_TYPES_AND_HALF(input_.scalar_type(),
+      "adaptive_max_pool2d_cuda",
+      [&] {
+        output.resize_({sizeB, sizeD, osizeH, osizeW});
+        indices.resize_({sizeB, sizeD, osizeH, osizeW});
+
+        scalar_t *input_data = input_.data<scalar_t>();
+        scalar_t *output_data = output.data<scalar_t>();
+        int64_t *indices_data = indices.data<int64_t>();
+
+        // cuda blocks & threads:
+        int blocksH = (int)(16L / sizeD);
+        blocksH = blocksH < 1 ? 1 : blocksH;
+        dim3 blocks(sizeB*sizeD, blocksH);
+        dim3 threads(32, 8);
+
+        // run maxpool kernel
+        adaptivemaxpool <<<blocks, threads, 0, at::cuda::getCurrentCUDAStream()>>> (
+                                   input_data, output_data,
+                                   indices_data,
+                                   isizeH, isizeW, osizeH, osizeW,
+                                   istrideD, istrideH, istrideW);
+      } 
+    );
+    THCudaCheck(cudaGetLastError());
```

Comment:
I think here you could change to `AT_CHECK(cudaGetLastError() == cudaSuccess, "Failed with error code ", cudaGetLastError())`

Reply:
I think `AT_CUDA_CHECK(cudaGetLastError())` is the most idiomatic, but we are also horribly inconsistent in native today as well and so I think it's not a big deal if it stays this way too.